File: composer/src/main/java/io/piano/android/composer/model/Access.java
package io.piano.android.composer.model;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

@Keep
public class Access {
    @NonNull
    @SerializedName("rid")
    public String resourceId = "";

    @NonNull
    @SerializedName("resourceName")
    public String resourceName = "";

    @SerializedName("expireDate")
    public int expireDate;

    @SerializedName("daysUntilExpiration")
    public int daysUntilExpiration;
}

File: composer/src/main/java/io/piano/android/composer/model/Access.kt
package io.piano.android.composer.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
public class Access(
    @JvmField
    @Json(name = "rid")
    public val resourceId: String = "",
    @JvmField public val resourceName: String = "",
    @JvmField public val expireDate: Int,
    @JvmField public val daysUntilExpiration: Int,
)
